### DISCUSSION TITLE
Monitoring / prometheus exporter

### DISCUSSION THREAD

**Discussion by jlpedrosa** [terraform-routeros/terraform-provider-routeros#390] [Ideas]
I've been having at ways to monitor miktorik in a way that is compatible with prometheus. I found https://github.com/akpw/mktxp but they don't seem to be actively maintaining the project. PRs not replied, basic crashes, ect. It is also written in Python.

@vaerh would you be interested in start together a mikrotik -> prometheus exporter?

**Comment by vaerh** [comment]
If I fit into your concept as second hands, I don't mind. I just don't have enough time periodically to maintain this project. For this reason, my cloud-init provider is not published in the HC registry. But I am quite capable of helping with writing code.
By the way, did you say about your project that it is in go? mktxp is fully on python.

**Reply by jlpedrosa** [reply]
Yeah what I meant is that I want the new project to be written in Go. They have a lot of open PRs that they don't reply to, users reporting crashes inside kube.

I think the size of the project is way smaller than the terraform provider.

**Comment by vaerh** [comment]
Why not? Let's give it a try.

**Comment by vaerh** [comment]
@jlpedrosa 
https://github.com/vaerh/mikrotik-stas-exporter

**Reply by jlpedrosa** [reply]
thank you! did you made me maintainer? also.... stas => sta**t**s , but probably the best name for that is mikrotik-prom-exporter.

In the prometheus communitiy, prometheus is shortened to "prom".

**Comment by vaerh** [comment]
Corrected. This is a PoC, so fix and rename whatever you want.

**Comment by vaerh** [comment]
You should receive an invitation from GitHub. If you have any problems, please write to me.

**Reply by jlpedrosa** [reply]
Yep, I'm in. BTW, IDK if there's a way for us to "chat"/"speak" synchornously ....

**Reply by jlpedrosa** [reply]
Also, BTW: https://www.reddit.com/r/mikrotik/comments/1bltz42/terraform_provider_mikrotik/

**Comment by vaerh** [comment]
If that's the question, there's a big problem I'm not using IM :(

OMG, to expect a user raid :D

**Comment by vaerh** [comment]
Meanwhile, HC tech support just ignored me. And I don't know what to do with the registry. Let's wait...

**Reply by jlpedrosa** [reply]
The new versions are out... https://registry.terraform.io/providers/terraform-routeros/routeros/latest not sure if you did something but they are unlocked, latest is available.

**Reply by vaerh** [reply]
Yes, I saw this weird thing and to see if it got better I made a new release. But it didn't publish.

**Reply by jlpedrosa** [reply]
nah, 1.41 is not here :(

**Comment by vaerh** [comment]
There were 2 problems there at once: no owner and artifact validation error. The first one was solved, I am still working on the second one.
